File: backend/src/main/java/com/personalsite/blog/entity/Practice.java
package com.personalsite.blog.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName(value = "practice", autoResultMap = true)
public class Practice extends BaseEntity {
    private String category;
    private String categoryIcon;
    private String name;
    private String description;
    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<PracticeLink> links;
    // todo | in_progress | mastered
    private String status;
    private Integer sortOrder;
}

File: frontend/src/types/practice.ts
export interface PracticeLink {
  title: string;
  url: string;
}

export interface Practice {
  id: number;
  category: string;
  categoryIcon: string;
  name: string;
  description: string;
  links: PracticeLink[];
  status: 'todo' | 'in_progress' | 'mastered';
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
}

export interface PracticeRequest {
  category: string;
  categoryIcon: string;
  name: string;
  description?: string;
  links: PracticeLink[];
  status: 'todo' | 'in_progress' | 'mastered';
  sortOrder: number;
}
